samza-core/src/main/java/org/apache/samza/coordinator/metadatastore/CoordinatorStreamStore.java

```diff
@@ -188,6 +188,14 @@ public void flush() {
     }
   }
 
+  @Override
```

Comment:
preferred way of calling super.finalize should be:
finalize() {
  try {
   .. stop...
} finally {
  super.finalize();
}
}